wom/src/main/scala/wom/graph/GraphNode.scala

```diff
@@ -97,7 +97,8 @@ object GraphNode {
     */
   trait GeneratedNodeAndNewNodes {
     def node: GraphNode
-    def newInputs: Set[_ <: GraphInputNode]
+    def newInputs: Set[_ <: ExternalGraphInputNode]
+    def nestedOuterGraphInputNodes: Set[_ <: OuterGraphInputNode]
```

Comment:
Looking at what these look like in ScatterNode I *think* I understand what they are but a comment would be helpful, also why they are useful.